processor/transformprocessor/internal/metrics/functions.go

```diff
@@ -15,10 +15,83 @@
 package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/metrics"
 
 import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
 	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
 )
 
+// registry is a map of names to functions for metrics pipelines
+var registry = map[string]interface{}{
+	"convert_sum_to_gauge": convertSumToGauge,
+	"convert_gauge_to_sum": convertGaugeToSum,
+}
+
+func init() {
+	// Init metrics registry with default functions common to all signals
+	for k, v := range common.DefaultFunctions() {
+		registry[k] = v
+	}
+}
+
 func DefaultFunctions() map[string]interface{} {
-	// No metric-only functions yet.
-	return common.DefaultFunctions()
+	return registry
+}
+
+func convertSumToGauge() (common.ExprFunc, error) {
+	return func(ctx common.TransformContext) interface{} {
+		mtc, ok := ctx.(metricTransformContext)
+		if !ok {
+			return nil
+		}
+
+		metric := mtc.GetMetric()
+		if metric.DataType() != pmetric.MetricDataTypeSum {
+			return nil
+		}
+
+		dps := metric.Sum().DataPoints()
+
+		metric.SetDataType(pmetric.MetricDataTypeGauge)
+		// Setting the data type removed all the data points, so we must copy them back to the metric.
+		dps.CopyTo(metric.Gauge().DataPoints())
+
+		return nil
+	}, nil
+}
+
+func convertGaugeToSum(stringAggTemp string, monotonic bool) (common.ExprFunc, error) {
+	var aggTemp pmetric.MetricAggregationTemporality
+	switch stringAggTemp {
+	case "delta":
+		aggTemp = pmetric.MetricAggregationTemporalityDelta
+	case "cumulative":
+		aggTemp = pmetric.MetricAggregationTemporalityCumulative
+	default:
+		return nil, fmt.Errorf("unknown aggregation temporality: %s", stringAggTemp)
+	}
```

Comment:
With #10197 getting merged in the accessor for `metric.aggregation_temporality` now [converts to and from the protobuf's numeric definition of aggregation temporality](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/48c784cbe7a96d706fae1f9bd95bfda09eac20c8/processor/transformprocessor/README.md?plain=1#L18).  I know this function doesn't need to use that accessor, but I think we should be consistent.  Either everything should use the protobuf's numeric value or everything should use a string value.  I opted for the protobuf definition since that the guide for other fields, but I am up for arguments advocating for string.

Reply:
Yeah, saw this was merged last night.

My argument for string based values here is readability from a user perspective; I wouldn't know without looking at the proto what delta aggregation temporality maps to, for instance. IMO, the string is a lot clearer.

Reply:
Agreed, that readability is much better.  If we switch to using a string for aggregation temporality I would prefer the string usage everywhere, so this switch statement would be moved somewhere it could be used by the accessor as well.  I also wish `pdata` had a string interpretation we could take advantage of instead of making our own.  The current implementation of `.String()` returns the proto's enum name which I wouldn't want to use over the numeric value

```go
var AggregationTemporality_name = map[int32]string{
	0: "AGGREGATION_TEMPORALITY_UNSPECIFIED",
	1: "AGGREGATION_TEMPORALITY_DELTA",
	2: "AGGREGATION_TEMPORALITY_CUMULATIVE",
}
```

@anuraaga @bogdandrutu what do you think?

Reply:
If we go with a string instead of the number we'll have to document that clearly because the readme and [processing.md doc](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/processing.md#telemetry-query-language) set the expectation that the protobuf/pdata is how fields are accessed/interacted with.